x/evm/keeper/grpc_query.go

```diff
@@ -333,9 +333,11 @@ func (k Keeper) EstimateGas(c context.Context, req *types.EthCallRequest) (*type
 		return nil, status.Error(codes.Internal, "failed to load evm config")
 	}
 
-	// ApplyMessageWithConfig expect correct nonce set in msg
-	nonce := k.GetNonce(ctx, args.GetFrom())
-	args.Nonce = (*hexutil.Uint64)(&nonce)
+	if args.Nonce == nil {
+		// ApplyMessageWithConfig expect correct nonce set in msg if not specify in request
+		nonce := k.GetNonce(ctx, args.GetFrom())
+		args.Nonce = (*hexutil.Uint64)(&nonce)
```

Comment:
[hermes](https://github.com/devashishdxt/hermes/pull/7/files) bump up nonce when call estimate

Reply:
isn't the nonce returned by `k.GetNonce` is already correct?

Reply:
nonce could be the same when estimate batches, 2nd batch call before 1st batch hasn't been finalize.

Reply:
is it possible to describe the test case in integration test outside of the ibc context, still confused what are you trying to do.

Reply:
just make it easier for me to compare between client and server, otherwise I need compare with data field ;D